### PR TITLE
Mtc migration tweaks

### LINKED ISSUE #418: ProjectSettings component has invalid messages
## Observed behavior (please include a screenshot if possible)

screen shot 2019-02-13 at 7 16 53 pm

## Expected behavior

There shouldn't be any places with messages that don't resolve to something.

## Steps to reproduce the problem

Try to create a user in the admin interface.

## Any special notes on configuration used

Used latest dev config.

## Version of datatools-ui and datatools-server if applicable (exact commit hash or branch name)

`dev` branch

### PR DESCRIPTION
**Pull request by landonreed**
This PR fixes a few issues discovered during the MTC migration to the RDBMS-backed Data Tools. In particular, these fixes are:

- not requesting deployments if the module is not enabled.
- fixing the toggle for GTFS feeds searchable in the alerts module.
- better handling of no results for GTFS graphQL request.
- fixes some broken things in the GTFS+ editor.
- fixes #418

Works with conveyal/datatools-server#168

### CODE REVIEW

File: lib/gtfsplus/actions/gtfsplus.js [hunk 1]
```diff
@@ -173,107 +175,104 @@ export function uploadGtfsPlusFeed (feedVersionId: string, file: File) {
...
     return uploadFile({file, url, token: user.token})
-      .then(result => {
-        console.log(result)
-        return dispatch(uploadedGtfsPlusFeed())
-      })
+      .then(result => dispatch(uploadedGtfsPlusFeed()))
   }
 }
 
+type GTFSPlusEntity = {
```
Comment (evansiroky) [inline lib/gtfsplus/actions/gtfsplus.js]: This is a kind of random placement.  I think it'd be a good convention to have all flow type definitions occur immediately after imports and just before the rest of the code.

File: lib/gtfsplus/actions/gtfsplus.js [hunk 2]
```diff
@@ -173,107 +175,104 @@ export function uploadGtfsPlusFeed (feedVersionId: string, file: File) {
...
+  rowIndex: number,
+  origRowIndex: number,
+  stop_id?: string,
+  route_id?: string,
+  [string]: any
```
Comment (evansiroky) [inline lib/gtfsplus/actions/gtfsplus.js]: This `[string]: any` is way too broad and allows for anything.  Please replace this with something more specific.

File: lib/gtfs/components/GtfsMap.js [hunk 3]
```diff
@@ -185,41 +173,6 @@ export default class GtfsMap extends Component<Props, State> {
     }
   }
 
-  renderIsochrones (isochroneBand: ?number, version: ?FeedVersion) {
```
Comment (evansiroky) [inline lib/gtfs/components/GtfsMap.js]: +1 for removing no longer needed code

File: lib/gtfs/util/index.js [hunk 4]
```diff
@@ -207,13 +207,12 @@ export function getEntityTableString (type: string): string {
  */
 export function searchEntitiesWithString (
```
Comment (evansiroky) [inline lib/gtfs/util/index.js]: Good candidate for adding a unit test for.

File: lib/gtfsplus/actions/gtfsplus.js [hunk 5]
```diff
@@ -173,107 +175,104 @@ export function uploadGtfsPlusFeed (feedVersionId: string, file: File) {
...
+      .then(results => dispatch(receiveGtfsEntities(results)))
+  }
+}
 
-    // determine which routes, stops, etc. aren't currently in the gtfsEntityLookup table and need to be loaded from the API
-    const routesToLoad = []
-    const stopsToLoad = []
-
-    const currentLookup = getState().gtfsplus.gtfsEntityLookup
+function lookupMissingEntities (tableId: string, rows: Array<GTFSPlusEntity>, currentLookup: Object) {
```
Comment (evansiroky) [inline lib/gtfsplus/actions/gtfsplus.js]: This is a great candidate for making a unit test for.

File: lib/gtfsplus/components/GtfsPlusTable.js [hunk 6]
```diff
@@ -19,6 +19,11 @@ export type Props = GtfsPlusEditorProps & {
   tableValidation: Array<any>
 }
 
+export const VISIBILITY = Object.freeze({
```
Comment (evansiroky) [inline lib/gtfsplus/components/GtfsPlusTable.js]: Add some comments here regarding what these constants mean.

File: lib/gtfsplus/containers/ActiveGtfsPlusEditor.js [hunk 7]
```diff
@@ -43,12 +43,15 @@ const mapStateToProps = (state: AppState, ownProps: Props) => {
   const feedSource = project && project.feedSources
     ? project.feedSources.find(fs => fs.id === feedSourceId)
     : null
-
+  const feedVersion = feedSource && feedSource.feedVersions
```
Comment (evansiroky) [inline lib/gtfsplus/containers/ActiveGtfsPlusEditor.js]: Could a selector be used to get this data?

File: lib/manager/actions/status.js [hunk 8]
```diff
@@ -210,11 +211,15 @@ export function startJobMonitor (showMonitor: boolean = true) {
     const timerFunction = () => dispatch(checkJobStatus())
```
Comment (evansiroky) [inline lib/manager/actions/status.js]: This function doesn't look like it's needed anymore.  Couldn't the dispatch be used directly?

Comment (landonreed) [reply lib/manager/actions/status.js]: Yea, I suppose we might reduce obfuscation by eliminating the `timerFunction`.

File: lib/gtfsplus/containers/ActiveGtfsPlusEditor.js (hunk 7 above)
Comment (landonreed) [reply lib/gtfsplus/containers/ActiveGtfsPlusEditor.js]: I'm not sure. We would need `feedVersionId` in the selector, but I don't know if there is a great way to do that.

**Comment by evansiroky** [comment]
When a new GTFS+ is published, the number of versions in the FeedVersionNavigator don't appear to get updated until the page is refreshed.

File: lib/gtfsplus/actions/gtfsplus.js (hunk 5 above)
Comment (landonreed) [reply lib/gtfsplus/actions/gtfsplus.js]: I have simplified this function quite a bit and added comments. Hopefully, that should be sufficient for improving maintainability at the moment.

**Comment by landonreed** [comment]
Thanks for the comments, @evansiroky. I haven't added any tests and wasn't able to confirm the issue you experienced with GTFS+ publishing not incrementing the version number. I have resolved everything else though and have simplified some of the functions that tests were recommended for (which I realize is not the same as adding tests). Let me know if you think this is enough to approve.

**Comment by evansiroky** [comment]
The issue with the feed version navigator updates to the latest number of versions (https://github.com/conveyal/datatools-ui/pull/382#issuecomment-459943759) has been fixed!

File: lib/common/containers/App.js [hunk 9]
```diff
@@ -20,9 +20,6 @@ import ActiveProjectsList from '../../manager/containers/ActiveProjectsList'
 import ActiveProjectViewer from '../../manager/containers/ActiveProjectViewer'
 import ActiveUserHomePage from '../../manager/containers/ActiveUserHomePage'
 import CreateProject from '../../manager/containers/CreateProject'
-import ActivePublicFeedSourceViewer from '../../public/containers/ActivePublicFeedSourceViewer'
-import ActivePublicFeedsViewer from '../../public/containers/ActivePublicFeedsViewer'
-import ActiveSignupPage from '../../public/containers/ActiveSignupPage'
```
Comment (evansiroky) [inline lib/common/containers/App.js]: RIP public pages

File: lib/alerts/components/CreateAlert.js [hunk 10]
```diff
@@ -28,7 +30,7 @@ export default class CreateAlert extends Component<Props> {
         bsSize='large'
         style={{marginTop: '-8px'}}
         disabled={createDisabled}
-        onClick={createAlert}
```
Comment (evansiroky) [inline lib/alerts/components/CreateAlert.js]: Shouldn't there be a linting error now that the `createAlert` is no longer used in the `render` method?

Comment (landonreed) [reply lib/alerts/components/CreateAlert.js]: Yes, I think you're right. And I have no idea why that commit would have done anything... Perhaps when I merged dev into this branch something got all funky, but that doesn't really explain the contents of the commit or the skipped lint.

**Pull request merged**